### DISCUSSION TITLE
Does sycl can support cuda tensorcore invoking？

### DISCUSSION THREAD

**Discussion by wangzy0327** [intel/llvm#11635] [Q&A] [answered]
Does sycl can support cuda tensorcore invoking？
If support，how to invoke cuda tensorcore?

**Comment by JackAKirk** [comment, marked as answer]
Yes, DPC++ supports the same Tensor Cores capabilities as CUDA Runtime API (minus experimental ones). The full list is here https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc#nvidia-tensor-cores-supported-combinations
and that document tells you how to use them within the `joint_matrix` dpc++ extension.

**Reply by JackAKirk** [reply]
Also basic tests for this functionality are available in this test file (for Tensor Cores operations supported by sm80 and later): https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/Matrix/joint_matrix_tensorcores_sm80.cpp
See the joint_matrix_tensorcores_sm72.cpp and joint_matrix_tensorcores_sm70.cpp files for Tensor Cores operations supported by earlier sm versions.

**Reply by wangzy0327** [reply]
@JackAKirk Does AMD matrix core support at present？

**Reply by JackAKirk** [reply]
No but I expect the PR to add it to be merged very soon: https://github.com/intel/llvm/pull/11485 , although note this is only for gfx90a gpus.
